emData/generate_MC.py

```diff
@@ -0,0 +1,186 @@
+
+from __future__ import absolute_import
+from __future__ import print_function
+import os,sys, re
+
+TF_index = ['L1L2', 'L2L3', 'L3L4', 'L5L6', 'D1D2', 'D3D4', 'L1D1', 'L2D1']
+TF_index = {k:v for v,k in enumerate(TF_index)}
+
+def ASRegion(region):
+  if region in ['L1', 'L2', 'L3']: return 'BARRELPS'
+  elif region in ['L4', 'L5', 'L6']: return 'BARREL2S'
+  else: return 'DISK'
+
+def APRegion(region):
+  if region in ['L1', 'L2', 'L3']: return 'BARRELPS'
+  elif region in ['L4', 'L5', 'L6']: return 'BARREL2S'
+  else: return 'DISK'
+
+def FMRegion(region):
+  if region in ['L1', 'L2', 'L3', 'L4', 'L5', 'L6']: return 'BARREL'
+  else: return 'DISK'
+
+if len(sys.argv) < 2:
+    print("Usage: " + sys.argv[0] + " WIRES_FILE")
+    sys.exit(1)
+wiresFileName = sys.argv[1]
+
+# First, parse the wires file and store the memory names associated with MCs in
+# dictionaries with the MC names as keys.
+wiresFile = open(wiresFileName)
+CMMems = {}
+FMMems = {}
+for line in wiresFile:
+    # Only barrel-only seeds are supported right now.
+    # Works for MC L3 phi sector C
+    if not any(["MC_" + l + "PHIB" in line for l in ["L3", "L4", "L5", "L6"]]) and \
+       not any(["MC_" + l + "PHIC" in line for l in ["L3"]]): #MC_L3PHIC was used for testing
+        continue
+    line = line.rstrip()
+    tcName = re.sub(r".*MC_(......).*", r"MC_\1", line)
+    memName = line.split()[0]
+    if memName.startswith("CM_"):
+        if tcName not in CMMems:
+            CMMems[tcName] = []
+        CMMems[tcName].append(memName)
+    if memName.startswith("FM_"):
+        FM = re.sub(r"FM_(....)_......", r"\1", memName)
+        if tcName not in FMMems:
+            FMMems[tcName] = []
+        FMMems[tcName].append(FM)
+wiresFile.close()
+
+# Open and print out preambles for the parameters and top files.
+dirname = os.path.dirname(os.path.realpath('__file__'))
+parametersFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculator_parameters.h"), "w")
+parametersFile.write(
+    "#ifndef TrackAlgorithm_MatchCalculator_parameters_h\n"
+    "#define TrackAlgorithm_MatchCalculator_parameters_h\n"
+    "\n"
+    "// This file contains numbers of memories and bit masks that are specific to\n"
+    "// each MatchCalculator and that come directly from the wiring.\n"
+    "//\n"
+    "// The validity of each of the barrel CM memories is determined by\n"
+    "// FMMask. The bits of this mask, from least significant to most\n"
+    "// significant, represent the memories in the order they are passed to\n"
+    "// MatchCalculator; e.g., the LSB corresponds to\n"
+    "// TF::L1L2. If a bit is set, the corresponding memory is\n"
+    "// valid, if it is not, the corresponding memory is not valid.\n"
+)
+topHeaderFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculatorTop.h"), "w")
+topHeaderFile.write(
+    "#ifndef TrackAlgorithm_MatchCalculatorTop_h\n"
+    "#define TrackAlgorithm_MatchCalculatorTop_h\n"
```

Comment:
@bryates Please change `TrackAlgorithm` → `TrackletAlgorithm`

Reply:
Done. I'll push the change soon.